### PR TITLE
Fix LR decay in TF Trainer

### PR DESCRIPTION
**Pull request by jplu**
This PR fixes mainly the issue #5045. It also provides a better alignment with the PT trainer with:

- a `set_seed()` function
- use the `logging_first_step` argument
- better logging message when training and load from checkpoint

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
This is great! LGTM.

File: src/transformers/trainer_tf.py [hunk 1]
```diff
@@ -333,18 +363,23 @@ def train(self) -> None:
...
-                if self.global_step % self.args.logging_steps == 0:
+                if (
+                    self.global_step % self.args.logging_steps == 0
+                    or self.global_step == 1
+                    and self.args.logging_first_step
+                ):
                     logs = {}
                     logs["loss"] = training_loss.numpy()
                     logs["learning_rate"] = lr_scheduler(self.global_step).numpy()
                     logs["epoch"] = self.epoch_logging
```
Comment (borisdayma) [inline src/transformers/trainer_tf.py:374]: I think this line can be removed too

Comment (jplu) [reply src/transformers/trainer_tf.py]: Indeed! Just removed them.

**Comment by jplu** [comment]
Just rebase on master, should be ok to merge now, unless you have some other things you want me to change @LysandreJik ?

**Pull request closed without merging**